src/ElevatorBooking.tsx

```diff
@@ -0,0 +1,572 @@
+      selectedDate?.getMonth() || new Date().getMonth(),
+      selectedDate?.getDate() || new Date().getDate(),
+      setTime.getHours(),
+      setTime.getMinutes(),
+    );
+
+    setSelectedEndDateChange(roundToMinuteInterval(endDate, 15));
+  };
+
+  const useStyles = makeStyles((theme: Theme) => createStyles({
+    root: {
+      '& .MuiTextField-root': {
+        margin: theme.spacing(1),
+        width: '100%',
+      },
+    },
+    registerButton: {
+      backgroundColor: '#f37f30',
+      color: 'white',
+      marginBottom: '20px',
+    },
+    paper: {
+      position: 'absolute',
+      width: 400,
+      backgroundColor: theme.palette.background.paper,
+      border: '2px solid #000',
+      boxShadow: theme.shadows[5],
+      padding: theme.spacing(2, 4, 3),
+    },
+  }));
+
+  const classes = useStyles();
+
+  return (
+    <div>
+      { thanks && (
+        <div className="section flex-grow">
+          <Grid container spacing={1}>
+            <Grid item xs={12}>
+              <h4 className="center">Amenity reserved</h4>
+              <p className="center">
+                Thank you!
+                {'  '}
+                <span role="img" aria-label="">🤟</span>
+                Your reservation has been confirmed!
+                {'  '}
+              </p>
+            </Grid>
+            <Grid item xs={12} className="center">
+              <Button
+                variant="contained"
+                className={classes.registerButton}
+                onClick={(): void => {
+                  setAvailability(null);
+                  setThanks(false);
+                  setSelectedStartDateChange(roundToMinuteInterval(new Date(), 15));
+                  setSelectedEndDateChange(roundToMinuteInterval(addMinutes(selectedStartDate, 30), 15));
+                }}
+                startIcon={<EventAvailable />}
+                type="submit"
+              >
+                Make Another Reservation
+              </Button>
+            </Grid>
+            <Grid item xs={12} className="center">
+              <Button
+                variant="contained"
+                onClick={(): void => navigate('/myreservations')}
+                className={classes.registerButton}
+                startIcon={<Schedule />}
+                type="submit"
+              >
+                My Reservations
+              </Button>
+            </Grid>
+          </Grid>
+        </div>
+      )}
+      { !thanks && (
+        <form className={classes.root} noValidate autoComplete="off" onSubmit={reserve}>
+          <div className="section flex-grow">
+            <Grid container spacing={5}>
+              <Grid item xs={12}>
+                <h4 className="center">Reserve an Amenity</h4>
+                { errorMessage && (
+                  <Alert severity="error">
+                    <AlertTitle>Error</AlertTitle>
+                    {errorMessage}
+                  </Alert>
+                )}
+              </Grid>
```

Comment:
Should have more input fields for unit, owner type, names, phones, deposit? (how is this handled - does the condo decide this value?) and move type.

Reply:
In the same sense, are there multiple elevators to book? If not, then I believe the dropdown for reservations should be removed/fixed.

Reply:
This will be finished in a different PR... just super rough scaffolding for now.